### ISSUE TITLE
Intl support

### ISSUE THREAD

**Issue by kyleconroy** [bytecodealliance/StarlingMonkey#45]
How difficult would it be to add support for the [Intl](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl) namespace object?

**Comment by guybedford** [comment]
We are currently hitting this LLVM issue on this build - https://github.com/llvm/llvm-project/issues/100084.

**Comment by kyleconroy** [comment]
@guybedford looks like there is a resolution of the LLVM issue?

**Comment by guybedford** [comment]
Yes, that issue is resolved, so that this is now blocked on the Spidermonkey upgrade path for StarlingMonkey.

**Comment by kyleconroy** [comment]
Yikes, that sounds scary. Is there currently a plan for that?

**Comment by guybedford** [comment]
It will come, but there are no immediate plans unfortunately.

**Comment by andreiltd** [comment]
> that this is now blocked on the Spidermonkey upgrade path for StarlingMonkey

Is this about moving away from forked SpiderMonkey and inlining the streams implementation in StarlingMonkey?

**Comment by guybedford** [comment]
This is rebasing to the latest Spidermonkey, see the previous rebase PR at https://github.com/bytecodealliance/gecko-dev/pull/50 (the PBL changes have since landed upstream).

**Comment by cfallin** [comment]
> (the PBL changes have since landed upstream).

Well, *most* of them -- I have to get to the last part of the weval-specific patches and submit them upstream if we don't want to drop weval support or rebase it carefully. Unfortunately I'm needing to prioritize other work at the moment and there's no clear timeline for me to get to this (perhaps in the latter half of 2025).

**Comment by cfallin** [comment]
A bit of an update: I was able to spend a today today working on isolating and putting together a patch for the last bits of weval support to upstream (current version on work-in-progress branch [here](https://github.com/cfallin/gecko-dev/commit/0052bcced52fc7803f979501e99886ff50f298a2)). It is hitting some errors during wevaling and while I had a day to work on this, I don't have unlimited time to debug at the moment, so I'll need to let it sit for a little bit more. Hopefully at some point I'll get another chunk of time and try to push this forward more.
